api/src/routes/anime/animeRoutes.ts

```diff
@@ -0,0 +1,15 @@
+import express from "express";
+import { animeController } from "@@root/controllers";
+import { extractPaginationParams } from "@@root/middleware";
+
+const animeRoutes = express.Router();
+
+animeRoutes.route("/anime").get(extractPaginationParams, animeController.index);
+
+animeRoutes.route("/anime/:id")
+    .post(animeController.create)
+    .get(animeController.read)
```

Comment:
Any of the controller that use pagination or `res.locals.params` need middleware to be passed before the controller reference.

You can import then with `import { extractPaginationParams, validation } from '@@middleware/index.js';`

For `id` validation you can use it like this:
```ts
animeRoutes.route("/anime/id/:id")
.get(validation.id, animeController.read)
```

For routes/methods that need pagination you can do:
```ts
animeRoutes.route("/")
.get(extractPaginationParams, animeController.index)
```